controllers/hcpvaultsecretsapp_controller.go

```diff
@@ -288,6 +299,58 @@ func (r *HCPVaultSecretsAppReconciler) Reconcile(ctx context.Context, req ctrl.R
 	}, nil
 }
 
+func (r *HCPVaultSecretsAppReconciler) startShadowSecretCleanupRoutine(ctx context.Context) {
+	logger := log.FromContext(ctx)
+	// cleanup of orphaned shadow secrets runs once every hour
+	ticker := time.NewTicker(1 * time.Hour)
+
+	// this process is expected to run indefinitely
+	for {
+		select {
+		case <-ticker.C:
+			err := r.cleanupOrphanedShadowSecrets(ctx)
+			if err != nil {
+				logger.Error(err, "Failed to cleanup orphaned shadow secrets")
+			}
+		}
+	}
+}
+
+func (r *HCPVaultSecretsAppReconciler) cleanupOrphanedShadowSecrets(ctx context.Context) error {
+	logger := log.FromContext(ctx)
+
+	// get all secrets in the operator namespace
+	secrets := corev1.SecretList{}
+	if err := r.List(ctx, &secrets, client.InNamespace(common.OperatorNamespace)); err != nil {
+		logger.Error(err, "Failed to list secrets")
+	}
+
+	for _, secret := range secrets.Items {
+		namespace := secret.Labels[hvsaLabelPrefix+"/namespace"]
+		name := secret.Labels[hvsaLabelPrefix+"/name"]
+
+		o := &secretsv1beta1.HCPVaultSecretsApp{}
+
+		// get the HCPVaultSecretsApp instance that that the shadow secret belongs to (if applicable)
+		err := r.Get(ctx, types.NamespacedName{Namespace: namespace, Name: name}, o)
+		if err != nil {
+			logger.Error(err, "Error getting resource from k8s", "secret", secret.Name)
+			continue
+		}
+
+		// delete the HCPVaultSecretsApp if it no longer exists
+		if o.GetDeletionTimestamp() != nil {
+			if err := r.handleDeletion(ctx, o); err != nil {
+				return err
+			}
+
+			logger.Info("Deleted orphaned resources associated with HCPVaultSecretsApp", "app", o.Name)
+		}
```

Comment:
It seems like this will need to handle the case where the Get() error is just that the object isn't found ([example here](https://github.com/hashicorp/vault-secrets-operator/blob/857159f9578a48c07d25158b0d9f65ba0897efb3/controllers/hcpvaultsecretsapp_controller.go#L118)). It's entirely possible that the HCPVaultSecretsApp is gone from k8s, but the hvs-dynamic-secret-cache secret is leftover, in which case there'd be nothing to call GetDeletionTimestamp() on.

If the HCPVaultSecretsApp is found, we should also compare the UID label ("vso-ownerRefUID") value to see if it matches the UID of the HCPVaultSecretsApp. If they don't match only the secret should be deleted.